pkg/scheduler/dependency_graph.go

```diff
@@ -381,11 +426,8 @@ func (sched *Scheduler) allocateReplicas(flow *client.Flow, options interfaces.D
 	}
 
 	sort.Stable(&tmp)
-	var result []string
-	for _, replica := range tmp.Items {
-		result = append(result, replica.ReplicaName())
-	}
-	return result[:count], nil
+
+	return tmp.Items[:count], nil, nil
 }
 
 // BuildDependencyGraph loads dependencies data and creates the DependencyGraph
```

Comment:
I see. The problem is that this part is mostly doing variable initialization and those variables are used later in the function (resdefs, dependencies etc.) If I'd extract this code into separate function this would be a function that does several unrelated things (loads dependencies, resdefs, checks arguments etc.) each of them is a single-liner + error check  and thus cannot be simplified. If we move all of them to a separate function, what's the name of that function would be? One function that does several unrelated things and returns multiple values of different types doesn't look like simplification